### PR TITLE
add reverse dependencies check

### LINKED ISSUE #124: Add check for orphaned add-on
Over time some add-ons might become broken/obsolete and will be removed. Some have a certain helper add-on which we then also forget to remove. Would be nice if we could detect this in some way if there are any. This can be done in the general scan on each branch and doesn't need to bother the regular submission report.

### PR DESCRIPTION
**Pull request by Rechi**
fixes #124

@MartijnKaijser is it correct to only warn for add-ons which have `script.module.*` as id?

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
That should suffice for now I think.cant think of anything else

**Comment by MartijnKaijser** [comment]
Shouldn't you add license header for the new files?

**Comment by Rechi** [comment]
rebased on top of #139 and added license headers for the new files
needs another rebase after #139 gets merged

File: kodi_addon_checker/addons/Repository.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from io import BytesIO
+
+
+class Repository(object):
+    """docstring for Repository"""
+    def __init__(self, version, path):
+        super(Repository, self).__init__()
+        self.version = version
+        self.path = path
+        gz_file = requests.get(path, timeout=(10, 10)).content
```
Comment (enen92) [inline kodi_addon_checker/addons/Repository.py]: maybe you can check the header first instead of assuming it is gzip

Comment (Rechi) [reply kodi_addon_checker/addons/Repository.py]: This requests an explicit gz file and doesn't assume auto compression by the web server.
https://github.com/xbmc/addon-check/blob/b803624f41a8d0e84629063f3fc886d86ae0f458/kodi_addon_checker/check_addon.py#L29

**Pull request merged**